File: init-db.py
#!/usr/bin/env python3
"""
SQLite Database Initialization Script
Initializes the database with all tables and dummy product data

Usage: python3 init-db.py
"""

import sqlite3
import os
import sys
from pathlib import Path

def init_database():
    # Get paths
    db_path = Path(__file__).parent / 'database' / 'database.sqlite'
    sql_path = Path(__file__).parent / 'database' / 'init-database.sql'
    
    print(f'[Database Init] Starting database initialization...')
    print(f'[Database Init] Database path: {db_path}')
    print(f'[Database Init] SQL file path: {sql_path}')
    
    # Check if SQL file exists
    if not sql_path.exists():
        print(f'[Database Init] ERROR: SQL file not found at {sql_path}')
        return False
    
    # Read the SQL file
    try:
        with open(sql_path, 'r') as f:
            sql = f.read()
        print('[Database Init] SQL file loaded successfully')
    except Exception as e:
        print(f'[Database Init] ERROR: Failed to read SQL file: {str(e)}')
        return False
    
    # Connect to database
    try:
        conn = sqlite3.connect(str(db_path))
        cursor = conn.cursor()
        print('[Database Init] Connected to SQLite database')
        
        # Enable foreign keys
        cursor.execute('PRAGMA foreign_keys = ON')
        
        # Execute SQL script
        cursor.executescript(sql)
        conn.commit()
        print('[Database Init] ✓ Database initialized successfully!')
        print('[Database Init] ✓ All tables created')
        print('[Database Init] ✓ Dummy products inserted (24 products across 6 categories)')
        
        # Verify the data
        cursor.execute("SELECT COUNT(*) as count FROM products")
        product_count = cursor.fetchone()[0]
        print(f'[Database Init] ✓ Total products in database: {product_count}')
        
        cursor.execute("SELECT COUNT(*) as count FROM categories")
        category_count = cursor.fetchone()[0]
        print(f'[Database Init] ✓ Total categories in database: {category_count}')
        
        print('[Database Init] ✓ Ready to start the application!')
        print('[Database Init] ✓ Run: php artisan serve')
        
        conn.close()
        return True
        
    except Exception as e:
        print(f'[Database Init] ERROR: Failed to initialize database: {str(e)}')
        return False

if __name__ == '__main__':
    success = init_database()
    sys.exit(0 if success else 1)

File: init-db.js
#!/usr/bin/env node

/**
 * SQLite Database Initialization Script
 * Runs the init-database.sql file to populate the database with tables and dummy data
 * 
 * Usage: node init-db.js
 */

const sqlite3 = require('sqlite3').verbose();
const fs = require('fs');
const path = require('path');

const dbPath = path.join(__dirname, 'database', 'database.sqlite');
const sqlPath = path.join(__dirname, 'database', 'init-database.sql');

console.log('[Database Init] Starting database initialization...');
console.log(`[Database Init] Database path: ${dbPath}`);
console.log(`[Database Init] SQL file path: ${sqlPath}`);

// Check if SQL file exists
if (!fs.existsSync(sqlPath)) {
    console.error(`[Database Init] ERROR: SQL file not found at ${sqlPath}`);
    process.exit(1);
}

// Read the SQL file
let sql;
try {
    sql = fs.readFileSync(sqlPath, 'utf8');
    console.log('[Database Init] SQL file loaded successfully');
} catch (err) {
    console.error('[Database Init] ERROR: Failed to read SQL file:', err.message);
    process.exit(1);
}

// Connect to database
const db = new sqlite3.Database(dbPath, (err) => {
    if (err) {
        console.error('[Database Init] ERROR: Failed to connect to database:', err.message);
        process.exit(1);
    }
    console.log('[Database Init] Connected to SQLite database');
    
    // Execute SQL script
    db.exec(sql, (err) => {
        if (err) {
            console.error('[Database Init] ERROR: Failed to execute SQL:', err.message);
            db.close();
            process.exit(1);
        }
        
        console.log('[Database Init] ✓ Database initialized successfully!');
        console.log('[Database Init] ✓ All tables created');
        console.log('[Database Init] ✓ Dummy products inserted (24 products across 6 categories)');
        
        // Verify the data
        db.get("SELECT COUNT(*) as count FROM products", (err, row) => {
            if (!err && row) {
                console.log(`[Database Init] ✓ Total products in database: ${row.count}`);
            }
            
            db.get("SELECT COUNT(*) as count FROM categories", (err, row) => {
                if (!err && row) {
                    console.log(`[Database Init] ✓ Total categories in database: ${row.count}`);
                }
                
                console.log('[Database Init] ✓ Ready to start the application!');
                console.log('[Database Init] ✓ Run: php artisan serve');
                
                db.close((err) => {
                    if (err) {
                        console.error('[Database Init] Warning: Error closing database:', err.message);
                    }
                    process.exit(0);
                });
            });
        });
    });
});
